pkg/router/template/router.go

```diff
@@ -99,7 +100,9 @@ type templateData struct {
 	// the directory that files will be written to, defaults to /var/lib/containers/router
 	WorkingDir string
 	// the routes
-	State map[string]ServiceUnit
+	State map[string](ServiceAliasConfig)
+	// the service lookup
+	ServiceUnits map[string]ServiceUnit
```

Comment:
Ok, just make sure that you're covered under the same locks Ram just
added.  And make sure you don't mutate any of the units, just copy
them before setting them into the map again.